<a href="https://colab.research.google.com/github/carrascalkevyn-sudo/Prog_CiviL_2026_1/blob/main/2_3_4_Actividad_Pluma_de_Agua_Algoritmo_GitHub.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#NOMBRES COMPLETOS: Kevyn Antonio Carrascal Pastor
# Codigo: 1103742196


# Actividad de Programación con Python  
## **Historia, ciencia y programación: la “pluma de agua”**

> **Relato de partida**  
> **¿Por qué los costeños le decimos “pluma” a la llave del agua?**  
>  
> En el Caribe colombiano, abrir la “pluma” no es solo una acción cotidiana: es una expresión cargada de historia, técnica y cultura. Aunque algunos se sorprendan o incluso se burlen del término, lo cierto es que decir “pluma” en vez de “llave” no es un error, sino una herencia viva que conecta nuestro lenguaje con siglos de evolución hidráulica.  
>  
> Mucho antes de que existieran los grifos modernos, el agua se distribuía mediante sistemas regulados con precisión. En España, por ejemplo, se usaba la **“pluma de agua”** como unidad de medida para calcular el caudal que se entregaba a cada hogar. Aproximadamente, una pluma equivalía a **0.025 litros por segundo**. En algunos lugares, ese flujo se regulaba mediante orificios calibrados y dispositivos llamados **plumeros**, que permitían controlar la salida del agua.  
>  
> Durante la época colonial, muchas de estas prácticas técnicas cruzaron el Atlántico y llegaron a ciudades como Cartagena, Barranquilla y Santa Marta. Con el tiempo, el término “pluma” dejó de ser solamente técnico y pasó al habla cotidiana como sinónimo de la llave del agua.  
>  
> Así, una palabra del uso diario conserva una memoria histórica, tecnológica y cultural.  
>  
> ---
>  
> **Idea central para esta actividad:**  
> A partir de este relato, construiremos un problema de programación donde una magnitud histórica y física, el **caudal de una pluma de agua**, se represente mediante variables, operaciones y estructuras básicas de Python.

## 1. Propósito de la actividad

Al finalizar esta actividad, el estudiante debe ser capaz de:

- interpretar un contexto histórico y científico como un problema computacional;
- identificar entradas, procesos y salidas;
- diseñar un algoritmo antes de programar;
- implementar el programa en Python;
- publicar el trabajo en GitHub;
- ampliar la documentación del código;
- clonar el repositorio al disco local usando Git.

> **Importante:**  
> En este notebook **no se entrega la solución completa**.  
> Aquí se presenta el **planteamiento**, la **estructura del problema** y el **algoritmo guía** para que cada estudiante construya el código.

## 2. Situación problema

Un hogar recibe una cierta cantidad de **plumas de agua**.  
Se sabe que:

- **1 pluma = 0.025 litros por segundo**

Con esta información, el estudiante debe construir un programa que permita calcular:

1. el caudal total en litros por segundo;
2. el consumo diario en litros;
3. el consumo mensual en litros (suponiendo 30 días);
4. una clasificación del consumo:
   - **Bajo**: menos de 1000 litros por día
   - **Medio**: entre 1000 y 3000 litros por día
   - **Alto**: más de 3000 litros por día

---

## 3. Análisis del problema

### 3.1 Entrada del sistema
El programa debe recibir:

- número de plumas asignadas al hogar.

### 3.2 Proceso
El programa debe:

- convertir el número de plumas en caudal total;
- transformar el caudal a consumo diario;
- transformar el consumo diario a consumo mensual;
- clasificar el nivel de consumo.

### 3.3 Salida
El programa debe mostrar:

- caudal total;
- consumo diario;
- consumo mensual;
- clasificación del consumo.

## 4. Variables y constantes sugeridas

El estudiante puede trabajar con variables como las siguientes:

- `plumas`
- `caudal_pluma`
- `caudal_total`
- `segundos_dia`
- `consumo_diario`
- `consumo_mensual`
- `nivel_consumo`

> Puede cambiar los nombres, pero deben ser claros y significativos.

## 5. Algoritmo propuesto

Antes de programar, escriba el algoritmo en lenguaje natural o seudocódigo.

### Algoritmo en lenguaje natural

1. Solicitar al usuario el número de plumas de agua.
2. Definir el valor de una pluma en litros por segundo.
3. Definir la cantidad de segundos que tiene un día.
4. Calcular el caudal total.
5. Calcular el consumo diario.
6. Calcular el consumo mensual.
7. Determinar si el consumo es bajo, medio o alto.
8. Mostrar los resultados de manera clara.

### Seudocódigo guía

```text
Inicio
    Leer número de plumas
    Definir caudal de una pluma
    Definir segundos por día

    Calcular caudal total
    Calcular consumo diario
    Calcular consumo mensual

    Si consumo diario es menor que 1000
        nivel = "Bajo"
    Si no, si consumo diario está entre 1000 y 3000
        nivel = "Medio"
    En otro caso
        nivel = "Alto"

    Mostrar caudal total
    Mostrar consumo diario
    Mostrar consumo mensual
    Mostrar nivel
Fin
```

## 6. Espacio para que el estudiante construya la solución

### Instrucciones
Complete el siguiente bloque de código a partir del algoritmo anterior.

- No copie una solución terminada sin comprenderla.
- Escriba comentarios propios.
- Use nombres de variables comprensibles.
- Verifique que los resultados tengan sentido físico.

In [1]:
# ==========================================================
# ACTIVIDAD: MODELAR LA "PLUMA DE AGUA" CON PYTHON
# ==========================================================

# 1. Solicitar al usuario el número de plumas
plumas = float(input("Ingrese el número de plumas: "))

# 2. Definir constantes del problema
caudal_pluma = 0.025  # litros por segundo
segundos_dia = 24 * 60 * 60  # segundos en un día

# 3. Calcular el caudal total
caudal_total = plumas * caudal_pluma

# 4. Calcular el consumo diario
consumo_diario = caudal_total * segundos_dia

# 5. Calcular el consumo mensual
consumo_mensual = consumo_diario * 30  # Suponiendo 30 días en el mes

# 6. Clasificar el consumo
if consumo_diario < 1000:
    nivel_consumo = "Bajo"
elif 1000 <= consumo_diario <= 3000:
    nivel_consumo = "Medio"
else:
    nivel_consumo = "Alto"

# 7. Mostrar resultados
print(f"\n--- Resultados ---")
print(f"Caudal total: {caudal_total:.4f} litros/segundo")
print(f"Consumo diario: {consumo_diario:.2f} litros")
print(f"Consumo mensual: {consumo_mensual:.2f} litros (estimado para 30 días)")
print(f"Clasificación del consumo: {nivel_consumo}")

Ingrese el número de plumas: 10

--- Resultados ---
Caudal total: 0.2500 litros/segundo
Consumo diario: 21600.00 litros
Consumo mensual: 648000.00 litros (estimado para 30 días)
Clasificación del consumo: Alto


## 8. Extensiones Opcionales Implementadas

Basado en las sugerencias de la celda 8, he implementado las siguientes extensiones:

1.  **Cálculo del Consumo Semanal.**
2.  **Conversión de Consumo a Metros Cúbicos.**
3.  **Presentación de resultados en formato tabular** (DataFrame de Pandas) para una mejor legibilidad.

In [ ]:
import pandas as pd

# 1. Calcular el consumo semanal
consumo_semanal = consumo_diario * 7

# 2. Convertir litros a metros cúbicos
# 1 metro cúbico (m³) = 1000 litros
consumo_diario_m3 = consumo_diario / 1000
consumo_semanal_m3 = consumo_semanal / 1000
consumo_mensual_m3 = consumo_mensual / 1000

# 3. Presentar los resultados en forma tabular (DataFrame)
data = {
    'Métrica': [
        'Caudal Total (L/s)',
        'Consumo Diario (Litros)',
        'Consumo Diario (m³)',
        'Consumo Semanal (Litros)',
        'Consumo Semanal (m³)',
        'Consumo Mensual (Litros)',
        'Consumo Mensual (m³)',
        'Clasificación del Consumo'
    ],
    'Valor': [
        f'{caudal_total:.4f}',
        f'{consumo_diario:.2f}',
        f'{consumo_diario_m3:.2f}',
        f'{consumo_semanal:.2f}',
        f'{consumo_semanal_m3:.2f}',
        f'{consumo_mensual:.2f}',
        f'{consumo_mensual_m3:.2f}',
        nivel_consumo
    ]
}

df_resultados = pd.DataFrame(data)
display(df_resultados)

## 7. Verificación conceptual

Antes de dar por terminado el programa, responda en comentarios o en una celda Markdown:

1. ¿Qué representa físicamente la variable principal del problema?
2. ¿Por qué este ejercicio conecta historia, cultura e ingeniería?
3. ¿Qué operaciones aritméticas fueron necesarias?
4. ¿Qué estructura condicional utilizó y por qué?
5. ¿Cómo interpretaría el resultado en un contexto real?

## Respuestas a las Preguntas de Verificación Conceptual

1.  **¿Qué representa físicamente la variable principal del problema?**
    La variable principal de entrada es `plumas`. Físicamente, representa el número de "plumas de agua" asignadas a un hogar, que es una unidad histórica y culturalmente relevante para medir el caudal de agua en algunas regiones. Una pluma equivale a 0.025 litros por segundo.

2.  **¿Por qué este ejercicio conecta historia, cultura e ingeniería?**
    Este ejercicio conecta historia y cultura a través del uso del término "pluma de agua" y su significado histórico en el Caribe colombiano para medir el flujo de agua. Se conecta con la ingeniería al traducir esta unidad histórica en una magnitud física (litros por segundo) y al aplicar cálculos para determinar el consumo de agua (caudal, consumo diario y mensual), que son conceptos fundamentales en la ingeniería hidráulica y la gestión de recursos.

3.  **¿Qué operaciones aritméticas fueron necesarias?**
    Las operaciones aritméticas necesarias fueron principalmente la **multiplicación**. Se utilizaron para:
    *   Calcular el `caudal_total` (número de plumas * caudal por pluma).
    *   Calcular `segundos_dia` (horas * minutos * segundos).
    *   Calcular el `consumo_diario` (caudal total * segundos en un día).
    *   Calcular el `consumo_mensual` (consumo diario * número de días en un mes).

4.  **¿Qué estructura condicional utilizó y por qué?**
    Se utilizó una estructura condicional `if-elif-else` para clasificar el consumo de agua (`nivel_consumo`). Esta estructura es adecuada porque permite evaluar múltiples condiciones en secuencia (Bajo, Medio, Alto) y ejecutar un bloque de código específico tan pronto como se cumple una condición, garantizando que solo una clasificación sea asignada al consumo diario.

5.  **¿Cómo interpretaría el resultado en un contexto real?**
    Interpretando el resultado del ejemplo (10 plumas):
    *   **Caudal total: 0.2500 litros/segundo.** Esto significa que el hogar tiene una capacidad constante de flujo de 0.25 litros de agua por cada segundo.
    *   **Consumo diario: 21600.00 litros.** Este es un volumen considerable de agua para un solo día. En un contexto real, podría indicar un consumo muy elevado, posiblemente excesivo para un hogar promedio, lo que podría traducirse en facturas de agua altas y un impacto significativo en el recurso.
    *   **Consumo mensual: 648000.00 litros.** Esto refuerza la idea de un consumo extremadamente alto, equivalente a aproximadamente 648 metros cúbicos al mes.
    *   **Clasificación del consumo: Alto.** Esta clasificación advierte que el consumo excede el umbral de 3000 litros por día, señalando una posible necesidad de investigar las causas del alto uso, como fugas, hábitos de consumo ineficientes, o un uso industrial/comercial no declarado.

## 8. Extensión opcional

Si desea profundizar, puede ampliar el programa para que también permita:

- calcular el consumo semanal;
- comparar el consumo de varios hogares;
- estimar el costo del agua si se conoce una tarifa por litro o por metro cúbico;
- convertir litros a metros cúbicos;
- presentar los resultados en forma tabular.

> Estas extensiones no reemplazan la solución base; son una ampliación.

## 9. Documentación del código con apoyo de IA

Después de construir y probar su programa:

1. Copie su código en ChatGPT.
2. Solicite una **explicación técnica en Markdown** del código.
3. Pida además:
   - descripción de la función del programa;
   - explicación de cada sección;
   - explicación de variables, entradas, procesos y salidas;
   - observaciones sobre claridad y buenas prácticas.
4. Mejore su código con base en esa retroalimentación.
5. Actualice su repositorio en GitHub con la versión mejorada y documentada.

### Prompt sugerido para documentar el código

```text
Actúa como docente universitario experto en programación con Python.

Te voy a compartir un código en Python.
Necesito que expliques el código en formato Markdown de manera clara, técnica y ordenada.

Incluye:
1. propósito general del programa;
2. explicación de entradas, procesos y salidas;
3. descripción de cada variable importante;
4. explicación de las estructuras usadas;
5. observaciones sobre claridad, estilo y posibles mejoras;
6. una versión comentada del código, sin cambiar su lógica.

Aquí está mi código:
[PEGAR AQUÍ EL CÓDIGO]
```

## 9. Documentación del código con apoyo de IA

Como su asistente de IA, he generado la siguiente documentación para su código, siguiendo las directrices que proporcionó:

---

### Documentación del Código en Python

#### 1. Propósito General del Programa

El programa tiene como objetivo calcular y clasificar el consumo de agua de un hogar basándose en una unidad de medida histórica y cultural del Caribe colombiano conocida como "pluma de agua". Traduce esta unidad a magnitudes físicas de caudal y consumo (diario y mensual), y luego categoriza el nivel de consumo (Bajo, Medio, Alto) para proporcionar una estimación de su impacto.

#### 2. Explicación de Entradas, Procesos y Salidas

*   **Entradas:**
    *   El programa solicita al usuario que ingrese el `número de plumas` asignadas al hogar. Esta es la única entrada requerida para iniciar los cálculos.

*   **Procesos:**
    1.  **Definición de Constantes:** Se establecen valores fijos para el caudal de una pluma (`caudal_pluma`) y la cantidad de segundos en un día (`segundos_dia`).
    2.  **Cálculo del Caudal Total:** Multiplica el número de plumas ingresado por el caudal de una sola pluma para obtener el caudal total en litros por segundo.
    3.  **Cálculo del Consumo Diario:** Convierte el caudal total a un volumen de consumo en litros por día, multiplicándolo por la cantidad de segundos en un día.
    4.  **Cálculo del Consumo Mensual:** Estima el consumo mensual multiplicando el consumo diario por 30 (asumiendo un mes de 30 días).
    5.  **Clasificación del Consumo:** Utiliza una estructura condicional (`if-elif-else`) para determinar si el `consumo_diario` es "Bajo" (menos de 1000 litros), "Medio" (entre 1000 y 3000 litros) o "Alto" (más de 3000 litros).

*   **Salidas:**
    *   El programa imprime de forma clara y formateada el `caudal total` calculado, el `consumo diario`, el `consumo mensual` estimado y la `clasificación del consumo` (Bajo, Medio, Alto).

#### 3. Descripción de Variables Importantes

*   `plumas` (float): Cantidad de "plumas de agua" que tiene el hogar. Es la entrada del usuario.
*   `caudal_pluma` (float): Constante que define el caudal de una sola pluma en litros por segundo (0.025 L/s).
*   `segundos_dia` (int): Constante que representa el número total de segundos en un día (86400 segundos).
*   `caudal_total` (float): El caudal total de agua que fluye en litros por segundo, calculado a partir de `plumas` y `caudal_pluma`.
*   `consumo_diario` (float): Volumen total de agua consumida en un día, en litros.
*   `consumo_mensual` (float): Volumen total de agua consumida en un mes (30 días), en litros.
*   `nivel_consumo` (str): Clasificación del consumo de agua (Bajo, Medio, Alto) basada en el `consumo_diario`.

#### 4. Explicación de las Estructuras Usadas

*   **Función `input()`:** Utilizada para obtener la entrada del usuario (`plumas`).
*   **Función `float()`:** Convierte la entrada del usuario (que `input()` devuelve como cadena de texto) a un número de tipo flotante, permitiendo realizar operaciones aritméticas.
*   **Operaciones Aritméticas:** Se utilizan `*` (multiplicación) para los cálculos de caudal y consumo, y `-` (resta) para la clasificación en el rango.
*   **Estructura Condicional `if-elif-else`:** Empleada para evaluar múltiples condiciones y asignar una clasificación al `nivel_consumo` basándose en el `consumo_diario`. Permite un flujo de control ramificado.
*   **Función `print()`:** Se utiliza para mostrar los resultados al usuario en la consola.
*   **F-strings (formatted string literals):** Facilitan la inclusión de valores de variables dentro de cadenas de texto de salida, permitiendo un formato claro y legible (`f"Texto {variable:.2f}"`).

#### 5. Observaciones sobre Claridad, Estilo y Posibles Mejoras

*   **Claridad y Estilo:** El código es muy claro y sigue buenas prácticas, como el uso de nombres de variables descriptivos y comentarios concisos para explicar cada sección. El formato de salida es legible.
*   **Robustez:** Se podría mejorar la robustez añadiendo manejo de errores para la entrada del usuario (por ejemplo, qué pasa si el usuario ingresa texto en lugar de un número). Un bloque `try-except` alrededor de `float(input(...))` sería beneficioso.
*   **Modularidad:** Para programas más grandes, las secciones de cálculo y clasificación podrían encapsularse en funciones separadas para mejorar la reutilización y la legibilidad (e.g., `calcular_consumo_diario(plumas)`, `clasificar_consumo(consumo)`).
*   **Constantes:** Las constantes `caudal_pluma` y `segundos_dia` están bien definidas, lo que facilita su mantenimiento.

#### 6. Versión Comentada del Código

```python
# ==========================================================
# ACTIVIDAD: MODELAR LA "PLUMA DE AGUA" CON PYTHON
# ==========================================================

# 1. Solicitar al usuario el número de plumas
# Se pide al usuario que ingrese la cantidad de 'plumas' y se convierte a tipo flotante.
# Esto permite manejar números decimales para las plumas si fuera necesario.
plumas = float(input("Ingrese el número de plumas: "))

# 2. Definir constantes del problema
# Define el caudal que representa una sola 'pluma de agua' en litros por segundo.
caudal_pluma = 0.025  # litros por segundo
# Define la cantidad de segundos totales en un día (24 horas * 60 minutos * 60 segundos).
segundos_dia = 24 * 60 * 60  # segundos en un día

# 3. Calcular el caudal total
# Multiplica el número de plumas por el caudal de cada pluma para obtener el caudal total.
caudal_total = plumas * caudal_pluma

# 4. Calcular el consumo diario
# Calcula el volumen total de agua consumida en un día, en litros.
# (Caudal total en L/s * segundos en un día).
consumo_diario = caudal_total * segundos_dia

# 5. Calcular el consumo mensual
# Estima el consumo total de agua en un mes, asumiendo 30 días por simplicidad.
consumo_mensual = consumo_diario * 30  # Suponiendo 30 días en el mes

# 6. Clasificar el consumo
# Utiliza una estructura condicional if-elif-else para categorizar el consumo diario.
# "Bajo": menos de 1000 litros/día.
# "Medio": entre 1000 y 3000 litros/día (inclusive ambos límites).
# "Alto": más de 3000 litros/día.
if consumo_diario < 1000:
    nivel_consumo = "Bajo"
elif 1000 <= consumo_diario <= 3000:
    nivel_consumo = "Medio"
else:
    nivel_consumo = "Alto"

# 7. Mostrar resultados
# Imprime todos los resultados calculados al usuario de forma clara y formateada.
# Se usan f-strings para insertar las variables directamente y el formato :.2f o :.4f
# para controlar el número de decimales en la salida.
print(f"\n--- Resultados ---")
print(f"Caudal total: {caudal_total:.4f} litros/segundo")
print(f"Consumo diario: {consumo_diario:.2f} litros")
print(f"Consumo mensual: {consumo_mensual:.2f} litros (estimado para 30 días)")
print(f"Clasificación del consumo: {nivel_consumo}")
```

## 10. Trabajo con GitHub

Una vez terminado el programa:

1. Cree un repositorio en GitHub.
2. Suba el notebook o el archivo `.py`.
3. Haga un primer commit.
4. Después de ampliar la documentación del código, haga un segundo commit.
5. Copie la URL del repositorio y consérvela para la entrega.

### Ejemplo de estructura del repositorio

```text
actividad-pluma-agua/
│
├── README.md
├── pluma_agua.py
└── Actividad_Pluma_de_Agua_Algoritmo_GitHub.ipynb
```

## 11. Comandos Git sugeridos

> **Nota:** estos comandos deben ejecutarse en la terminal, Git Bash o PowerShell.

### 11.1 Inicializar el repositorio local

```bash
git init
git add .
git commit -m "Versión inicial de la actividad pluma de agua"
```

### 11.2 Conectar con GitHub

```bash
git branch -M main
git remote add origin URL_DEL_REPOSITORIO
git push -u origin main
```

### 11.3 Subir una segunda versión con mejor documentación

```bash
git add .
git commit -m "Se mejora documentación y comentarios del código"
git push
```

## 12. Clonar el repositorio en el disco local C:

Una vez que el repositorio esté en GitHub, copie su URL y ejecute:

```bash
cd /c
git clone URL_DEL_REPOSITORIO
```

Si está usando PowerShell en Windows, también puede hacerlo así:

```powershell
cd C:\
git clone URL_DEL_REPOSITORIO
```

### Ejemplo esperado

Si su repositorio se llama:

```text
https://github.com/usuario/actividad-pluma-agua.git
```

entonces el comando sería:

```bash
git clone https://github.com/usuario/actividad-pluma-agua.git
```

Con esto se creará una carpeta del proyecto dentro de `C:\`.

## 13. Evidencia de entrega sugerida

El estudiante debe entregar:

- el enlace del repositorio en GitHub;
- el notebook o archivo `.py` funcional;
- una versión documentada del código;
- evidencia de que el repositorio fue clonado en el equipo local.

### Evidencia mínima recomendada
- captura del repositorio en GitHub;
- captura de los commits;
- captura del comando `git clone` ejecutado correctamente.

## 14. Reflexión final

Este ejercicio no trata solamente de programar.  
También muestra cómo una palabra cotidiana del Caribe colombiano puede llevarnos a:

- la historia de la hidráulica;
- la representación computacional de una magnitud física;
- el diseño de algoritmos;
- el uso responsable de herramientas como GitHub y ChatGPT en el trabajo académico.

> La programación también puede nacer de la memoria cultural y del contexto local.

## 15. Espacio para notas del estudiante

Escriba aquí una breve reflexión personal:

- ¿Qué aprendió?
- ¿Qué parte le resultó más difícil?
- ¿Qué mejoró cuando documentó el código?
- ¿Qué importancia tiene publicar y versionar su trabajo?

## 15. Espacio para notas del estudiante (Reflexión personal)

### ¿Qué aprendió?
Aprendí a conectar un concepto histórico-cultural, como la "pluma de agua", con un problema de programación real. Esto me ayudó a entender cómo las unidades de medida, incluso las no estándar, pueden ser modeladas en código. También reforcé mis conocimientos sobre la importancia de la identificación de entradas, procesos y salidas, así como la definición de constantes y la aplicación de estructuras condicionales para la toma de decisiones dentro de un programa.

### ¿Qué parte le resultó más difícil?
La parte más difícil fue la interpretación inicial del contexto histórico y su traducción precisa a variables y cálculos en Python. Aunque el concepto de "pluma de agua" es interesante, asegurar que todos los factores (caudal por pluma, segundos en un día, etc.) se aplicaran correctamente para obtener resultados significativos fue el desafío principal.

### ¿Qué mejoró cuando documentó el código?
La documentación del código con apoyo de IA fue invaluable. Mejoró mi comprensión del propio código al obligarme a desglosarlo en sus componentes más fundamentales. Me permitió identificar claramente:
*   El propósito general del programa.
*   Las entradas, procesos y salidas de manera explícita.
*   La función de cada variable importante.
*   Las estructuras de control utilizadas y por qué eran apropiadas.
*   Puntos de mejora en cuanto a robustez (manejo de errores de entrada) y modularidad (uso de funciones).
*   Finalmente, la versión comentada del código es una guía excelente para futuras revisiones o para que otros entiendan rápidamente la lógica.

### ¿Qué importancia tiene publicar y versionar su trabajo?
Publicar y versionar el trabajo es fundamental por varias razones:
*   **Colaboración:** Permite que otros vean y contribuyan al proyecto, fomentando el aprendizaje mutuo.
*   **Historial de Cambios:** El control de versiones (Git/GitHub) crea un registro detallado de todas las modificaciones, lo que es esencial para rastrear errores, revertir cambios o entender la evolución del código.
*   **Portafolio:** Un repositorio bien documentado en GitHub sirve como un portafolio profesional, mostrando habilidades de programación y capacidad de gestión de proyectos.
*   **Acceso y Respaldo:** Proporciona un lugar seguro para almacenar el código, accesible desde cualquier lugar y actuando como un respaldo ante cualquier eventualidad local.